clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslClientAuthenticator.java

```diff
@@ -169,13 +187,17 @@ public void authenticate() throws IOException {
 
         switch (saslState) {
             case SEND_APIVERSIONS_REQUEST:
-                // Always use version 0 request since brokers treat requests with schema exceptions as GSSAPI tokens
-                ApiVersionsRequest apiVersionsRequest = new ApiVersionsRequest((short) 0);
-                send(apiVersionsRequest.toSend(node, nextRequestHeader(ApiKeys.API_VERSIONS, apiVersionsRequest.version())));
-                setSaslState(SaslState.RECEIVE_APIVERSIONS_RESPONSE);
-                break;
+                if (this.apiVersionsResponse == null) {
```

Comment:
This check is not needed if we set initial state during reauthentication to `SEND_HANDSHAKE_REQUEST`

Reply:
👍 removed